File: tree/binary_tree_right_side_view.py
# Q:199 - https://leetcode.com/problems/binary-tree-right-side-view/

from typing import Optional, List
from index import TreeNode

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        queue, result = [], []

        queue.append(root)

        while queue:
            pass


c = Solution()

# root1 = [1,2,3,null,5,null,4]
# root2 = [1,null,3]
# root3 = []
root1 = TreeNode(1, TreeNode(2, None, TreeNode(5)), TreeNode(3, None, TreeNode(4)))
root2 = TreeNode(1, None, TreeNode(3))
root3 = None


print(c.levelOrder(root1))
print(c.levelOrder(root2))
print(c.levelOrder(root3))

File: tree/binary_tree_right_side_view.js
// # Q:199 - https://leetcode.com/problems/binary-tree-right-side-view/

/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
 var rightSideView = function(root) {
    const result = []
    if (root === null){
        return result;
    }
    
    const queue = [root];
    
    
    while (queue.length) {
        const len = queue.length;
        const currentList = [];
        
        for (let i = 0; i < len; i++){
            const current = queue.shift();
        
            currentList.push(current.val)

            if (current.left){
                queue.push(current.left)
            }

            if (current.right){
                queue.push(current.right)
            }
        }
        
        result.push(currentList[currentList.length - 1])
        
    }
    
    return result;
};